src/poller.spec.ts

```diff
@@ -1,41 +1,260 @@
 import * as td from 'testdouble';
 
+import { POLL_INTERVAL_MS, POLL_JITTER_PCT } from './constants';
 import initPoller from './poller';
 
-describe('initPoller', () => {
-  const testInterval = 10;
-  const callback = td.func<() => Promise<void>>();
-  jest.useFakeTimers();
+describe('poller', () => {
+  const testIntervalMs = POLL_INTERVAL_MS;
+  const maxRetryDelay = testIntervalMs * POLL_JITTER_PCT;
+  const noOpCallback = td.func<() => Promise<void>>();
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
 
   afterEach(() => {
     td.reset();
     jest.clearAllTimers();
   });
 
-  it('starts polling at interval', async () => {
-    const poller = initPoller(testInterval, callback);
-    await poller.start();
-    td.verify(callback(), { times: 1 });
-    await verifyPoll(2);
-    await verifyPoll(3);
+  afterAll(() => {
+    jest.useRealTimers();
   });
 
-  it('stops polling', async () => {
-    const poller = initPoller(testInterval, callback);
-    await poller.start();
-    poller.stop();
-    jest.advanceTimersByTime(testInterval * 10);
-    await flushPromises();
```

Comment:
🪓  We did not need this `flushPromises()` nonsense; we only needed to use `await jest.advanceTimersByTimeAsync()` instead of the synchronous `jest.advanceTimersByTime()` that was being used.